lib/pbench/cli/server/shell.py

```diff
@@ -1,11 +1,12 @@
 #!/bin/env python3
 
-from pbench.server.api import create_app
+from pbench.server.api import App
 import subprocess
 
 
 def main():
-    app = create_app()
+    flask_app = App()
```

Comment:
I had to do this to avoid using the `global` variables in api. I put the method `create_app()` inside the class `App` and therefore to call this method I had to create an `App` object

Reply:
I don't get why you need to do any more than pick up `resource_class_args` to drop the `global app`; that's all I did.

Reply:
There are `global app` references in the other classes.  Those should be created passing the `app` object, no?

Reply:
yes, but I need to import `app`, `db`, `blacklist` variables in `db_models.py` file. Our database models class inherit from SQLAlchemy's `Model` class

Reply:
The DB model shouldn't depend on App. The model should be completely hidden under the user management class, and it shouldn't depend on anything outside that boundary.

Again, I'd really like to get all these "globals" out of the App and into user management, probably using a singleton object under your Resource(s). If your Resource needs to use its `app` reference to hang onto some persistent value, that's OK, but try to connect everything through that level?

Reply:
Yes, I found a better way to define the DB models, it still depends on the app (but class doesn't have to depend on it, we can pass the app via init) for password hashing and logging, but its much cleaner than the current implementation